src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrderingPartyController.cs

```diff
@@ -74,18 +76,64 @@ public async Task<ActionResult<OrderingPartyModel>> GetAsync(string orderId)
         }
 
         [HttpPut]
-        public ActionResult Update(string orderId, OrderingPartyModel model)
+        public async Task<ActionResult> UpdateAsync(string orderId, OrderingPartyModel model)
         {
-            if (orderId is null)
+            if (model is null)
+            {
+                throw new ArgumentNullException(nameof(model));
+            }
+
+            var order = await _orderRepository.GetOrderByIdAsync(orderId);
+            if (order is null)
             {
                 return NotFound();
             }
 
-            if (model is null)
+            var primaryOrganisationId = User.GetPrimaryOrganisationId();
+            if (primaryOrganisationId != order.OrganisationId)
             {
-                throw new ArgumentNullException(nameof(model));
+                return Forbid();
             }
 
+            var primaryContact = model.PrimaryContact;
+
+            var contact = new Contact
+            {
+                FirstName = primaryContact.FirstName,
+                LastName = primaryContact.LastName,
+                Email = primaryContact.EmailAddress,
+                Phone = primaryContact.TelephoneNumber
+            };
+
+            order.OrganisationContact = contact;
+
+            var organisationModel = model.Organisation;
+            var addressModel = organisationModel.Address;
+            var address = new Address
+            {
+                Line1 = addressModel.Line1,
+                Line2 = addressModel.Line2,
+                Line3 = addressModel.Line3,
+                Line4 = addressModel.Line4,
+                Line5 = addressModel.Line5,
+                Town = addressModel.Town,
+                County = addressModel.County,
+                Postcode = addressModel.Postcode,
+                Country = addressModel.Country
+            };
+
+            order.OrganisationName = organisationModel.Name;
+            order.OrganisationOdsCode = organisationModel.OdsCode;
+            order.OrganisationAddress = address;
+
+            var name = User.Identity.Name;

```

Comment:
See the class `ClaimsPrincipalExtensions` as we have an extension to retrieve the user name for consistency reasons.